### ISSUE TITLE
Formsy.Form onChange is fired twice

### ISSUE THREAD

**Issue by TuxujPes** [christianalfoni/formsy-react#203]
Hi, facing an issue with `Formsy.Form` fires `onChange` twice even though my `Formsy.Mixin.setValue` is invoked only once.
Might not be a big perf impact, but doesn't seem necessary either, or is it?

**Comment by Semigradsky** [comment]
I think that is the `onChange` event from the input. Just add `stopPropaganation`:
```js
  changeValue: function (event) {
    event.stopPropagation();
    this.setValue(event.currentTarget.value);
  },
```

**Comment by TuxujPes** [comment]
Nope, that didn't help. And i don't think it could, it reacts solely to `setValue`, no?
Well, at least, `this.isChanged()` second argument on second change is `false`, so i can check on that and prevent unnecessary store updates.

**Comment by Semigradsky** [comment]
I a little changed [`login` example](https://github.com/christianalfoni/formsy-react/tree/react-0.14/examples):

In `Input.js`:
```js
  changeValue(event) {
    console.log('Change!');
    this.setValue(event.currentTarget[this.props.type === 'checkbox' ? 'checked' : 'value']);
  },
```
In `app.js`:
```js
  onChange() {
    console.log('Changed!');
    console.log(arguments);
  },
  render() {
    return (
      <Form onChange={this.onChange} ...
```

I see in console:
![image](https://cloud.githubusercontent.com/assets/1198848/9788368/f5d1fb7c-57d1-11e5-984f-a85c1e3cd56e.png)


When I changed `changeValue` to:
```js
  changeValue(event) {
    event.stopPropagation();
    console.log('Change!');
    this.setValue(event.currentTarget[this.props.type === 'checkbox' ? 'checked' : 'value']);
  },
```
I see:
![image](https://cloud.githubusercontent.com/assets/1198848/9788406/3291ed10-57d2-11e5-9195-8a971ecdc6a7.png)

**Comment by Semigradsky** [comment]
If this does not help, can you show your code?

**Comment by TuxujPes** [comment]
My `Formsy.Form.onChange` does not get fired with `SyntheticEvent`, it is `(object, bool)` both times.

`input.js`
```js
onChange: function(e) {
  e.stopPropagation(); // having this here does not affect anything for me
  this.props.onChange(e.currentTarget.value); // fired once
}
```
`input_formsy.js`
```js
onChange: function(value) {
  this.setValue(value); // fired once
}
```
`some_form.js`
```js
onChange: function(data, isChanged) {
  actions.updateSomeForm(data); // fired twice
}
```

actually, i do not see this behaviour anymore... probably was some temp binding issue.. or whatever.
Anyway, thanks for the time spent on investigation.

**Comment by tomatau** [comment]
I have the same issue as the OP.

Using e.stopPropagation fixed the issue.

**Comment by sdemjanenko** [comment]
Im running into the same issue.  I think its because of the `render` function's `<form {...this.props} onSubmit={this.submit}>`.  The `props` are not having Formsy-only props filtered out (such as `onChange`).

**Comment by sdemjanenko** [comment]
I have a PR for a fix #258.
